cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/executor/ExecutionTaskPlanner.java

```diff
@@ -331,55 +329,56 @@ private void maybeAddLeaderChangeTasks(Collection<ExecutionProposal> proposals,
      * chances to make progress.
      */
     boolean newTaskAdded = true;
+    _interPartMoveBrokerIds.addAll(_interPartMoveTasksByBrokerId.keySet());
     Set<Integer> brokerInvolved = new HashSet<>();
     Set<TopicPartition> partitionsInvolved = new HashSet<>();
     while (newTaskAdded) {
       newTaskAdded = false;
+      // The first task of each brokerInvolved must changed. Let's remove the brokers, then add them again by comparing their new first tasks.
```

Comment:
Nit: `must changed` -> `might have changed`.
This is because, a broker B1 could be involved due to another broker B2's first task causing picking a non-first task from B1.